Update to Golang 1.13.9

justaugustus opened the pull request:
**What type of PR is this?**
/kind feature cleanup
/area dependency

**What this PR does / why we need it**:
- deps: Update to Golang 1.13.9
- build: Remove kube-cross image building

Accompanying kube-cross bump: https://github.com/kubernetes/release/pull/1200

/assign @BenTheElder @liggitt @cblecker @dims @listx
/milestone v1.18
cc: @kubernetes/release-team @kubernetes/release-engineering 
ref: [quarterly updates](https://github.com/kubernetes/release/issues/1134)

**Does this PR introduce a user-facing change?**:

```release-note
- deps: Update to Golang 1.13.9
- build: Remove kube-cross image building
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

justaugustus committed:
build: Remove kube-cross image building

kube-cross image building has moved to k/release/images/build/cross.

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus committed:
deps: Update to Golang 1.13.9

Signed-off-by: Stephen Augustus <saugustus@vmware.com>
Co-authored-by: Jeff Grafton <jgrafton@google.com>

justaugustus commented:
/retest

justaugustus commented:
/test pull-kubernetes-typecheck

justaugustus commented:
(Dropped the staging image commit (2ae4a0ed0e146e435984171cac4dffed415df064) now that the [kube-cross:v1.13.9-2 image has been promoted](https://github.com/kubernetes/k8s.io/pull/684))

justaugustus commented:
This PR is currently blocked on tests failing due to bazel.
I've provided details here: https://github.com/kubernetes/kubernetes/issues/88553#issuecomment-602411809

ixdy commented:
An alternative to updating rules_go is https://github.com/ixdy/kubernetes/commit/2a189ad34ee611a02a4bc1cda12deddcf1c15a44.

justaugustus commented:
@ixdy -- Thanks so much! I've squashed that in as b6179ac.

justaugustus commented:
/retest

justaugustus commented:
Looks like most of the tests are starting to pass, so I'm going to apply @ixdy's fix on the 1.14.1 PR (https://github.com/kubernetes/kubernetes/issues/88638) as well.

justaugustus committed:
build/dependencies: Remove bazel WORKSPACE go_version check

When supplying the go_download_sdk instead of updating rules_go, the
go_version is removed, so we shouldn't check for it anymore in
hack/verify-external-dependencies-version.sh.

Without this, pull-kubernetes-verify jobs will fail.

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus commented:
Added 979e3e0dd529bbcd0f0cdef22fc7f82f66ae48ce:
```
build/dependencies: Remove bazel WORKSPACE go_version check
    
When supplying the go_download_sdk instead of updating rules_go, the
go_version is removed, so we shouldn't check for it anymore in
hack/verify-external-dependencies-version.sh.

Without this, `pull-kubernetes-verify` jobs will fail.
```

liggitt commented on build/build-image/cross/VERSION:
```diff
@@ -1 +1 @@
-v1.13.8-1
+v1.13.9-2
```
this remains but the rest of the image is gone?

justaugustus replied:
@liggitt -- Yep. There are still a [few things that depend on it](https://cs.k8s.io/?q=cross%2FVERSION&i=nope&files=&repos=) being in that location.

So basically:
- production version: https://github.com/kubernetes/kubernetes/blob/master/build/build-image/cross/VERSION
- prerelease version: https://github.com/kubernetes/release/blob/master/images/build/cross/VERSION

I didn't want to introduce changing the references to this PR, but I am thinking about the best way to do it for a follow-up.

liggitt replied:
ok, as long as its on your radar

liggitt commented on build/root/WORKSPACE:
```diff
@@ -48,14 +48,31 @@ http_archive(
+        "linux_s390x": ("go1.13.9.linux-s390x.tar.gz", "a40949aaf55912b06df8fda511c33fde3e52d377706bdc095332652c1ad225e3"),
+        "windows_386": ("go1.13.9.windows-386.zip", "e22406377448f1aea2dd1517327e5ae452d826c0c7624b3511d5af510c57b69a"),
+        "windows_amd64": ("go1.13.9.windows-amd64.zip", "cf066aabdf4d83c251aaace14b57a35aafffd1fa67d54d907f27fb31e470a135"),
+    },
```
deep magic from the dawn of time

ixdy replied:
it's a little bit of https://github.com/kubernetes/kubernetes/commit/efb21a54f19d6f64f3272df5d3048bd68360dcea mixed with https://github.com/bazelbuild/rules_go/blob/master/go/private/sdk_list.bzl.

liggitt commented:
go bits look good. I'll take @ixdy's word on the bazel bits from https://github.com/ixdy/kubernetes/commit/2a189ad34ee611a02a4bc1cda12deddcf1c15a44

/lgtm
/approve

justaugustus commented:
Cherry picks opened:
- 1.18: https://github.com/kubernetes/kubernetes/pull/89398
- 1.17: https://github.com/kubernetes/kubernetes/pull/89399
- 1.16: https://github.com/kubernetes/kubernetes/pull/89400

justaugustus commented:
/test pull-kubernetes-e2e-gce
